core/utils/cuckoo_map.h

```diff
@@ -180,7 +180,7 @@ class CuckooMap {
   Entry* Find(const K& key, const H& hasher = H(), const E& eq = E()) {
     // Blame Effective C++ for this
     return const_cast<Entry*>(
-        static_cast<const typeof(*this)&>(*this).Find(key, hasher, eq));
+        static_cast<const CuckooMap&>(*this).Find(key, hasher, eq));
```

Comment:
Is it safe to do this? If a `DerivedClass` redefines (not overrides) the const version of `Find`, then it should invoke `DerivedClass::Find()`, not `CuckooMap::Find()`